### PR TITLE
New action domain preselection

### PR DESCRIPTION
**Pull request by chinins**
## Description

This PR adds preselection of domain & motion domain on relevant actions (Create payment, Edit Domain, Permission management) when clicking on New action with a different domain than root.

I've talked to the product and it's ok if sometimes the dialogs are disabled (e.g. when there is no reputation in the selected domain).

**New stuff** ✨

- add domain id preselection to the relevant dialogs

Resolves DEV-404

### CODE REVIEW

**Review by alicjakujawa — CHANGES_REQUESTED** [review]
Seems to working fine. 
* One thing to find out, if we should also preselect domain for transfer funds (from field especially)
* Should be good to go after addressing my small comments

File: src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx
Comment (chinins) [reply src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx:58]: Yes, the way it should work is that we never show Root in the EditDomain. So it's just to preserve the same logic.

If we have a preselected domain other than Root or nothing, then we should show that domain. If not, then we show any other domain than Root which might be no domain at all.

File: src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx
Comment (chinins) [reply src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx:104]: I am afraid I need to search once again to get the name. I can of course do it in different places but this will have to happen twice (also only when it's root or no domain). 

Maybe I'm not seeing something? I am open to your suggestions.

File: src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx
Comment (chinins) [reply src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx:51]: good catch - leftover from testing! :)

**Comment by chinins** [comment]
@alicjakujawa, I was just waiting for confirmation from Karol for the TransferFunds dialog and it's all included now.

Please, see my answers to the rest of your comments.

File: src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx
Comment (alicjakujawa) [reply src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx:99]: what about something like that:
  const selectedDomain = useMemo(
    () => {
      const filteredDomains = domains.filter(({ethDomainId}) => ethDomainId !== ROOT_DOMAIN_ID);
      const domain = filteredDomains.find(({ethDomainId}) => ethDomainId === preselectedDomainId);
      return domain ? domain : filteredDomains[0];
    },
    [preselectedDomainId, domains],
  );

Then you at least dont have additional loop outside of memo function. Of course you would need to adjust it to the rest of things

Comment (chinins) [reply src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx]: I think I had a better idea that we indeed only have one loop. Hope you'll like this option.

File: src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx
Comment (alicjakujawa) [inline src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx:69]: Okey so now youre doing almost same job twice. useMemo and everything which is inside, its not cost free. So if we already have "selectedDomain" we dont need to do almost same things for "selectedDomainId" Please refactor it to get rid of "selectedDomainId" function

Comment (chinins) [reply src/modules/dashboard/components/EditDomainDialog/EditDomainDialog.tsx:58]: i just forgot to remove the old function. It should be removed now.

**Review by alicjakujawa — APPROVED** [review]
Now it seems to be good to go 🔥

File: src/modules/dashboard/components/TransferFundsDialog/TransferFundsDialog.tsx
Comment (rdig) [inline src/modules/dashboard/components/TransferFundsDialog/TransferFundsDialog.tsx:149]: Leftover ?

**Review by rdig — APPROVED** [review]
Works as expected, and the code looks good.

Very nice work! :100:

File: src/modules/dashboard/components/TransferFundsDialog/TransferFundsDialog.tsx
Comment (chinins) [reply src/modules/dashboard/components/TransferFundsDialog/TransferFundsDialog.tsx:149]: good catch, removed!

**Pull request merged**
